src/types/instantsearch.ts

```diff
@@ -153,7 +178,20 @@ export type RouteState = {
   [stateKey: string]: any;
 };
 
-export type SearchClient = Pick<
-  AlgoliaSearchClient,
-  'search' | 'searchForFacetValues'
->;
+export type SearchClient = Pick<Client, 'search' | 'searchForFacetValues'>;
```

Comment:
I feel like these additions could be on their own `algoliasearch` file, so some more explanation on the reasons for the complicated code can be explained. Now it seem to make this file more complicated

Reply:
You mean a definitions file inside `instantsearch.js`?

Reply:
I mean all the definitions related to the client probably should go in an adjacent `algoliasearch.ts` file, since it's not per se related to the instantsearch constructor